### ISSUE TITLE
name tag for all metrics configs

### ISSUE THREAD

**Issue by amoore877** [spiffe/spire#2114]
m3 config allows setting a service name tag for metrics: https://github.com/spiffe/spire/blob/v0.12/pkg/common/telemetry/m3.go#L56

this can be especially helpful in cases of multiple deployment flavors of SPIRE.

1. this is not a documented config option for m3 config https://github.com/spiffe/spire/blob/v0.12/doc/telemetry_config.md
2. we should introduce similar functionality as able to the other metric sink types. Ideally the various metric sinks should have as much parity in features as possible.

**Comment by azdagron** [comment]
Is this still something you'd find valuable in your deployments? It's been a few years and we've not seen any other requests for this.

**Comment by amoore877** [comment]
Yes I think this would add some ease of maintenance for operators in future solutions addressing our scaling wrt quickly filtering down observability data points. In present day though would not add anything.

**Comment by mpoegel** [comment]
I'm also interested in being able to customize the prefix for the metric names in my deployment. Is help still wanted for this? I'd like to contribute this feature if it'd be welcome.

**Comment by rturner3** [comment]
Hi @mpoegel, this issue is about being able to configure the value of a dedicated `service` label to all metrics when using M3 as the telemetry sink. Today the default values for the `service` label are `spire_agent` for the agent and `spire_server` for the server.

Are you looking to prefix M3 metric names with a prefix or are you using a different metrics sink?

**Comment by mpoegel** [comment]
Perhaps prefix isn't the right word here (I'm coming from Envoy, which refers to this setting as the [prefix](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/metrics/v3/stats.proto#extension-envoy-stat-sinks-statsd)). I'm specifically interested in the statsd sink, which also involves being able to configure the `serviceName` field that defaults to `spire_agent` and `spire_server`. For statsd and other sinks, that's [here](https://github.com/spiffe/spire/blob/main/pkg/common/telemetry/metrics.go#L74) and as @amoore877 pointed out for the m3 sink it's in a different [spot](https://github.com/spiffe/spire/blob/main/pkg/common/telemetry/m3.go#L56). @amoore877 mentioned adding similar functionality to the other sinks, so I thought this issue appropriate.

I got this working locally for statsd by adding a configuration option to the Telemetry section like this
```
telemetry {
    prefix = "abc-spire-agent"

    Statsd = [
        { address = "127.0.0.1:8125" }
    ]
}
```
